packages/http-client-java/generator/http-client-generator-test/src/main/java/tsptest/enumservice/models/PriorityModel.java

```diff
@@ -82,7 +82,13 @@ public String toString() {
     @Generated
     @Override
     public boolean equals(Object obj) {
-        return Objects.equals(this.value, obj);
```

Comment:
Ok, so previous code is just comparing wrong thing?

Though I assume `this == obj` may actually work? As when equal, both should be the same item in VALUES map.
Not hurt writing it this way, though.

Reply:
Yeah, previous code is wrong. 
Emm, guess we can use `==`. If user really initialize by using the private constructor, they are at their own risks.

Reply:
changed

Reply:
For `ExpandableEnum`, we used `ConcurrentHashMap` and `computeIfAbsent` in `fromValue`. Guess racing condition shouldn't happen?https://github.com/microsoft/typespec/blob/a9ac77db2f8c49c545d57a2b961211b8a20d4ff1/packages/http-client-java/generator/http-client-generator-test/src/main/java/tsptest/enumservice/models/PriorityModel.java#L46-L53

Reply:
For `ExpandableStringEnum`, we probably can't do that, since the constructor is public...

Reply:
Guess we could even remove the `get` and only do `computeIfAbsent`. Though having a get is more efficient for future gets.

Reply:
I am not sure whether there be multiple call of the lambda in `computeIfAbsent` upon race condition. The lambda may be called, then discarded if found the item already there (didn't check the source -- I kind of guess there is no guarantee on the call of lambda, only the guarantee in the items in Map).

Reply:
changed